homeassistant/components/evohome/__init__.py

```diff
@@ -119,13 +120,35 @@
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Create a (EMEA/EU-based) Honeywell TCC system."""
+    """Set up the Evohome integration from the configuration.yaml file."""
 
-    async def load_auth_tokens(store: Store) -> tuple[dict, dict | None]:
+    # until config flow is implemented, this most closely matches the current behaviour
+    if entries := hass.config_entries.async_entries(DOMAIN):
+        scan_interval: timedelta = config[DOMAIN].pop(CONF_SCAN_INTERVAL)
+        config[DOMAIN][CONF_SCAN_INTERVAL] = int(scan_interval.total_seconds())
+
+        hass.config_entries.async_update_entry(entries[0], data=config[DOMAIN])
+        return True
```

Comment:
This seems the best way - if the user disables, it stays disabled until they re-enable (even across restarts), and the config (`entry.data`) is updated only on restarts of HA.